samza-core/src/main/java/org/apache/samza/processor/StreamProcessor.java

```diff
@@ -269,10 +291,37 @@ public void stop() {
     }
   }
 
+  /**
+   * Get the {@link ProcessorContext} of this {@link StreamProcessor}
+   * @return the {@link ProcessorContext} object
+   */
+  public ProcessorContext getProcessorContext() {
```

Comment:
This is going to change as Cameron's shared context PR is coming along. I will remove it for now, since it is just a stub interface now.

Reply:
Another note on your comments on objects should be constructed and passed into StreamProcessor (i.e. following the DI pattern): currently, implementation of StreamProcessor does not allow that yet. e.g. JobCoordinator is constructed internally and processorId is only available after JobCoordinator is constructed, etc. I would prefer to have those refactor effort on StreamProcessor in a separate effort, since it is not a public API any more.

Reply:
Opened SAMZA-1833 to track this one.